torch/csrc/jit/autodiff.cpp

```diff
@@ -32,45 +32,46 @@ static std::vector<Value*> gradientForNode(Node* node, ArrayRef<Value*> grad_val
   const auto build_sym_grad = [node](const std::vector<SymbolicVariable>& grads) -> std::vector<SymbolicVariable> {
     auto inputs = fmap<SymbolicVariable>(node->inputs());
     auto outputs = fmap<SymbolicVariable>(node->outputs());
-    switch(node->kind()) {
-      case aten::add:
+	{
+	IR_IFC(node, aten::add)
```

Comment:
Case expressions must be constant and vc14 doesn't consider types constant.

Reply:
Yes, the fact that case expressions have to be constant expressions is part of the standard. `aten::add` is not a type, it's a `class enum` value, so it is a constant expression.

Reply:
Actually, might have been a casting issue? Not sure if kind is typed as enum or int. Maybe it would've worked if I tried something like case((int)aten::add)?